### PR TITLE
Adds Configuration IDs to case studies

### PR DESCRIPTION
**Pull request by vulder**
This allows user to persist and access configurations for a specific
case study.

related se-passau/VaRA#682

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- tests/base/test_configuration.py  1
- varats-core/varats/paper/case_study.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6300702&bid=20491680)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- varats-core/varats/paper/case_study.py  2
- tests/base/test_configuration.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6300702&bid=20491680)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- varats-core/varats/paper/case_study.py  2
- tests/base/test_configuration.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6300702&bid=20491680)

File: varats-core/varats/paper/case_study.py [hunk 1]
```diff
@@ -12,19 +13,37 @@
...
+        config_ids: tp.Optional[tp.List[int]] = None
+    ) -> None:
         self.__commit_hash = commit_hash
         self.__commit_id = commit_id
 
+        if config_ids:
+            self.__config_ids: tp.List[int] = config_ids
+        else:
+            # By default we add a list with the DummyConfig ID if no
+            # configurations where provided.
```
Comment (boehmseb) [inline varats-core/varats/paper/case_study.py]:
```suggestion
            # configurations were provided.
```

**Review by boehmseb — APPROVED** [review]

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- varats-core/varats/paper/case_study.py  2
- tests/base/test_configuration.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6300702&bid=20491680)

**Pull request merged**